crates/sortition/src/node_state.rs

```diff
@@ -0,0 +1,383 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+//
+// This file is provided WITHOUT ANY WARRANTY;
+// without even the implied warranty of MERCHANTABILITY
+// or FITNESS FOR A PARTICULAR PURPOSE.
+
+use actix::prelude::*;
+use alloy::primitives::U256;
+use anyhow::Result;
+use e3_data::{AutoPersist, Persistable, Repository};
+use e3_events::{
+    BusError, CiphernodeAdded, CiphernodeRemoved, CommitteePublished, ConfigurationUpdated,
+    EnclaveErrorType, EnclaveEvent, EventBus, OperatorActivationChanged, PlaintextOutputPublished,
+    Subscribe, TicketBalanceUpdated,
+};
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use tracing::info;
+
+/// State for a single ciphernode
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct NodeState {
+    /// Current ticket balance for this node
+    pub ticket_balance: U256,
+    /// Number of active E3 jobs this node is currently participating in
+    pub active_jobs: u64,
+    /// Whether this node is active (has met minimum requirements)
+    pub active: bool,
+}
+
+impl Default for NodeState {
+    fn default() -> Self {
+        Self {
+            ticket_balance: U256::ZERO,
+            active_jobs: 0,
+            active: false,
+        }
+    }
+}
+
+/// State for all nodes across all chains
+#[derive(Clone, Debug, Serialize, Deserialize, Default)]
+pub struct NodeStateStore {
+    /// Map of (chain_id, node_address) to node state
+    pub nodes: HashMap<(u64, String), NodeState>,
+    /// Current ticket price per chain
+    pub ticket_prices: HashMap<u64, U256>,
```

Comment:
Stupid question: do we need to be concerned about what token this is represented in?